.gitlab-ci.yml

```diff
@@ -205,107 +80,44 @@ publish:
           }
       } while ($i -lt 3)
 
-# The 2 jobs below, do not depend on the packaging stage. So they are always run, even if package stage can fail.
-generate-lib-init-tag-values:
-  tags: ["arch:amd64"]
-  image: registry.ddbuild.io/ci/auto_inject/gitlab:current
-  stage: deploy
-  rules:
-    - if: '$POPULATE_CACHE'
-      when: never
-    # We don't tag prerelease versions
-    - if: '$CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+$/'
-      when: always
-    - when: manual
-      allow_failure: true
-  variables:
-    IMG_DESTINATION_BASE: dd-lib-dotnet-init
-    ADDITIONAL_TAG_SUFFIXES: musl # comma separated list of additional tag suffixes
-  script:
-    - ./.gitlab/build-lib-init.sh
-  artifacts:
-    reports:
-      dotenv: build.env
-
-deploy-lib-init-trigger:
-  stage: deploy
-  trigger:
-#    project: DataDog/dd-trace-dotnet-gitlab-test # can be used for testing
-    project: DataDog/public-images
-    branch: main
-    strategy: depend
-  variables:
-    IMG_SOURCES: ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:$CI_COMMIT_SHA
-    IMG_DESTINATIONS: $IMG_DESTINATIONS
-    IMG_SIGNING: "false"
-  # needs the version from the generate-tag-values job
-  needs:
-    - job: generate-lib-init-tag-values
-      artifacts: true
-
-deploy-musl-lib-init-trigger:
-  stage: deploy
-  trigger:
-#    project: DataDog/dd-trace-dotnet-gitlab-test # can be used for testing
-    project: DataDog/public-images
-    branch: main
-    strategy: depend
-  variables:
-    IMG_SOURCES: ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:$CI_COMMIT_SHA-musl
-    IMG_DESTINATIONS: $IMG_DESTINATIONS_musl
-    IMG_SIGNING: "false"
-  # needs the version from the generate-tag-values job
-  needs:
-    - job: generate-lib-init-tag-values
-      artifacts: true
-
-package:
-  extends: .package
+download-single-step-artifacts:
+  stage: package
+  image: registry.ddbuild.io/docker:20.10.13-gbi-focal
+  tags: [ "arch:amd64" ]
   rules:
     - if: $DOTNET_PACKAGE_VERSION
       when: on_success
-    - if: '$CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(-prerelease)?$/'
+    - if: '$CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(-prerelease)?$/' # Manually triggered as artifacts are from the Github release
       when: manual
       allow_failure: false
+    - when: delayed # Artifacts come from Azure pipeline, wait a reasonable time before polling
+      start_in: 15 minutes
   script:
-    - ../.gitlab/build-deb-rpm.sh
-  variables:
-    ARCH: amd64
+    - .gitlab/download-single-step-artifacts.sh
+  artifacts:
+    expire_in: 2 weeks
+    paths:
+      - artifacts
 
-package-arm:
-  extends: .package-arm
-  rules:
-    - if: $DOTNET_PACKAGE_VERSION
-      when: on_success
-    - if: '$CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(-prerelease)?$/'
-      when: manual
-      allow_failure: false
-  script:
-    - ../.gitlab/build-deb-rpm.sh
+package-oci:
```

Comment:
Do these automatically extend the jobs of the same name that are in `one-pipeline.yml`? Out of interest, what's the difference in this vs explicitly using `extends`?

Reply:
the difference between `extends` and using the same name is that `extends` creates a new job where this overrides the previous job. In both cases, gitlab treats the `yaml` as a map/dictionary and merges the keys.

The reason I went for this style vs the previous `imported .package` + `package-oci extends .package` we used before is that the old way forced users to define `package-oci` even when there were no changes to be made. This way is less code in the default case.

Sidenote: there's also `<: *anchor` which is a worse version of `extends` that is built-in to YAML.